### PR TITLE
Correctly break during Png decoding

### LINKED ISSUE #2752: Loading image with DecoderOptions { MaxFrames = 1 } loads multiple frames for attached png file
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### ImageSharp version

3.1.4

### Other ImageSharp packages and versions

-

### Environment (Operating system, version and so on)

Windows 11

### .NET Framework version

net7.0

### Description

Loading an image with the following statement:
`    using var img = await SixLabors.ImageSharp.Image.LoadAsync(new DecoderOptions { MaxFrames = 1 }, fileStream);
`
should just load one frame, which is does for all images I have tested on, EXCEPT the attached one, where it 
imports all 42 frames.

### Steps to Reproduce

Load the attached "duck.png" like this:
```
  using var fileStream = File.OpenRead(@"c:\temp\duck.png");
  using var img = await SixLabors.ImageSharp.Image.LoadAsync(new DecoderOptions { MaxFrames = 1 }, fileStream);
  Assert.AreEqual(1, img.Frames.Count);
```

### Images

duck

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes #2752 

We weren't breaking from the loop only the nested switch. We were also missing a chunk break. The first `fcTL` chunk in a PNG is followed by one or more `IDAT` chunk, only subsequent `fcTL` chunks are followed by `fdAT` chunks.

The test image contains the following chunks. 

```
IHDR, pHYs, acTL, fcTL, IDAT ×3, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT,
fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, 
fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT,
fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT,
fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL,
fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT,
fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, fcTL, fdAT, fdAT, IEND
```

### CODE REVIEW

File: src/ImageSharp/Formats/Png/PngDecoderCore.cs [hunk 1]
```diff
@@ -271,6 +266,11 @@ public Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken
                                 previousFrameControl = currentFrameControl;
                             }
 
+                            if (frameCount == this.maxFrames)
```
Comment (brianpopow) [inline src/ImageSharp/Formats/Png/PngDecoderCore.cs:269]: maybe use `>= this.maxFrames` here to be extra safe?

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Png/PngDecoderCore.cs:269]: Yeah, multiple `fcTL` in a row might cause this. Will update.

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Png/PngDecoderCore.cs]: Done

**Review by brianpopow — APPROVED** [review]
looks good now

**Pull request merged**